src/rgw/rgw_op.cc

```diff
@@ -2214,6 +2215,16 @@ void RGWGetObj::execute(optional_yield y)
   std::unique_ptr<RGWGetObj_Filter> run_lua;
   map<string, bufferlist>::iterator attr_iter;
 
+  bool rgw_perf_counters_cache = s->cct->_conf.get_val<bool>("rgw_perf_counters_cache");
+  std::string labels = ceph::perf_counters::key_create(rgw_op_counters_key, {{"Bucket", s->bucket_name}, {"User", s->user->get_id().id}});
+  std::shared_ptr<PerfCounters> labeled_counters;
+  if(rgw_perf_counters_cache) {
+    labeled_counters = perf_counters_cache->get(labels);
+    if(labeled_counters) {
```

Comment:
for the rgw-specific interface, what do you think about an interface like this in rgw_perf_counters.h?
```c++
namespace rgw::op_counters {
  template <std::size_t Count>
  std::shared_ptr<PerfCounters> get(label_pair (&&labels)[Count]);
}
```
that way we could hide the global variables like `rgw_op_counters_key` and `rgw_perf_counters_cache` in rgw_perf_counters.cc. if the cache was disabled, it would return `nullptr`. then the ops would use it like this:
```c++
  auto labeled_counters = rgw::op_counters::get({
        {"Bucket", s->bucket_name},
        {"User", s->user->get_id().id}
      });
  if (labeled_counters) {
    labeled_counters->inc(l_rgw_labeled_get_ops);
  }
...
  if (labeled_counters) {
    labeled_counters->inc(l_rgw_labeled_get_b, end - ofs);
  }
...
```
if we also want to add the global instance of op counters, we could think about other interfaces that increment both the global counter and the labeled one if it exists:
```c++
namespace rgw::op_counters {
  void inc(PerfCounters* labeled, int idx, uint64_t v = 1) // labeled may be null
  {
    if (labeled) {
      labeled->inc(idx, v);
    }
    if (global_op_counters) {
      global_op_counters->inc(idx, v);
    }
  }
```
so the ops could update both with a single call, and not have to check whether `labeled_counters` is null:
```c++
  auto labeled_counters = rgw::op_counters::get({...});
  rgw::op_counters::inc(labeled_counters.get(), l_rgw_labeled_get_ops);
...
  rgw::op_counters::inc(labeled_counters.get(), l_rgw_labeled_get_b, end - ofs);
...
```